File: Python/kumarAbhi-kumar/1431_Kids_With_the_Greatest_Number_of_Candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        highestCandies = candies[0]
        for i in range(len(candies)):
            if highestCandies < candies[i]:
                highestCandies = candies[i]
        for i in range(len(candies)):
            candies[i] =  True if candies[i] + extraCandies >= highestCandies else False
        return candies

File: CPP/coolakt/1431. Kids With the Greatest Number of Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_candie=0;
        for(int i =0; i<candies.size(); i++){
            if(candies[i]>max_candie)
                max_candie=candies[i];
            
        }
        vector<bool> ans(candies.size());
        for(int j=0; j<candies.size(); j++){
            if(candies[j]+ extraCandies >= max_candie){
                ans[j]= true;
            }
            else{
                ans[j]=false;
            }
        }
        return ans;
    }
};
